Dynamo/Dynamo.SwiftLang/SLAttribute.cs

```diff
@@ -11,11 +11,22 @@ public SLAttribute (SLIdentifier name, CommaListElementCollection<SLBaseExpr> ar
 		{
 			Name = Exceptions.ThrowOnNull (name, nameof (name));
 			Add (new SimpleElememt ("@"));
+			StringRep = Name.Name;
 			Add (Name);
 			if (args != null) {
 				Add (new SimpleElememt ("("));
 				Add (args);
 				Add (new SimpleElememt (")"));
+				StringRep += "(";
+				foreach (var arg in args) {
+					try {
+						var argId = (SLIdentifier)arg;
+						StringRep += argId.Name;
+					} catch (Exception ex) {
```

Comment:
Under which conditions do exceptions occur? Is it possible to change the code so that it doesn't throw exceptions (exceptions can be exceptionally slow).

Reply:
The `try-catch` is meant to catch errors when casting. As far as I can tell, every arg in `args` is actually an `SLIdentifier`, but I didn't want to make that assumption without a guarantee hence the failsafe. @stephen-hawley any idea for a less expensive approach?

Reply:
I'd just do the cast and let it fail.

If this isn't an "impossible" case we'll get a crash and fix it to do type checking.

Reply:
You could check the type:

```csharp
if (arg is SLIdentifier argId)
    stringRep.Append (argId.Name);
```

Reply:
Ok I did just that in the latest change, fixed! Thanks @rolfbjarne @chamons